vsphere/internal/helper/virtualmachine/virtual_machine_helper.go

```diff
@@ -712,6 +713,22 @@ func Reconfigure(vm *object.VirtualMachine, spec types.VirtualMachineConfigSpec)
 	return task.Wait(tctx)
 }
 
+// UpgradeHardwareVersion attempts to upgrade the hardware version on the
+// specified virtual machine. The virtual machine must be powered off.
+func UpgradeHardwareVersion(vm *object.VirtualMachine, version int) error {
+	log.Printf("[DEBUG] Upgrading virtual machine %q", vm.InventoryPath)
+	ctx, cancel := context.WithTimeout(context.Background(), provider.DefaultAPITimeout)
+	defer cancel()
+	vm.UpgradeVM(ctx, GetHardwareVersionID(version))
```

Comment:
Should we be checking the current version of the VM before upgrading?  What if the VM has a later newer version or the same version?